### PR TITLE
[NETBEANS-837] Update winp to released 1.26 version and fix license

### PR DESCRIPTION
**Pull request by matthiasblaesing**
winp is actively developed at: https://github.com/kohsuke/winp

The code in recent versions of winp looks very similar to the state after the
patching. The API is stable (checked with ``japi-compliance-checker``), so
the upgrade should be save.

Part of the license problem that the license integration code is not prepared
for files showing up at build time (winp was build from source) and not coming
from the normal download route.

_Update: I intend to push this patch also to release90 when reviewed, as the
issue was raised for the upcoming 9.0 release._

### CODE REVIEW

**Comment by matthiasblaesing** [comment]
A point outside the PR itself: I think we should try to keep modifications to a minimum and active upstream releases should IMHO be preferred.
The patch lacks reasoning, so it is difficult to reason what was the reason it was introduced. I guess it introduced to fix a bug in an old version and then noone checked if it was fixed upstream.

**Comment by emilianbold** [comment]
Yes, of course, it's good to use upstream versions and only patch if necessary.

The history of the patch is convoluted.

It seems to me that initially it added the DDL_TARGET system property: https://github.com/emilianbold/netbeans-releases/commit/96cb331b988290db870e35450f5fcec0cdf6d74f#diff-ac12f5b0fbc0ffeb077ce213ec649449

And this is something winp still has: https://github.com/kohsuke/winp/blob/master/src/main/java/org/jvnet/winp/Native.java#L147

But then they had a memory leak https://netbeans.org/bugzilla/show_bug.cgi?id=180915 so they got rid of that system property and expressly ignore it: https://github.com/emilianbold/netbeans-releases/commit/77d8d691e17a546d2080b4a9e854f5240aff1431#diff-ac12f5b0fbc0ffeb077ce213ec649449

Now, I suspect that we might still get memory leaks with the current winp?

Maybe we have to clear the system property before calling winp?

We would need to test this on Windows before using the upstream version, see if #180915 still holds.

File: nbbuild/licenses/MIT-winp [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (emilianbold) [inline nbbuild/licenses/MIT-winp]: This file seems unnecessary, I've started a thread on dev@ to talk about this https://lists.apache.org/thread.html/b86dd173dd91f27d6e6b48c429f03bca5afe8e417fdbea4e9484da37@%3Cdev.netbeans.apache.org%3E

**Comment by matthiasblaesing** [comment]
I think the MIT license part is in-line with the other parts of netbeans. If we change it, that can be done later.

**For the patch:**
My interpretation is different (in fact the system property is used before and after patching, just differently):

It is visible in the current form of the patch:
https://github.com/apache/incubator-netbeans/blob/master/extexecution.process/external/winp.patch#L23

From my interpretation the patch changes the usage of the `winp.folder.preferred` system property. Before the patch the property changed the `jarFile` variable, afterwards it changes the `dllFile` variable. The patch is sensible, as the location of the extracted dll (this is held in `dllFile`) can so be adjusted through a system property. And this would also explain the memory inflation, if the dll is loaded multiple time with different names.

The current winp code shows exactly the fixed behavior.  The source if searched here:
https://github.com/kohsuke/winp/blob/master/src/main/java/org/jvnet/winp/Native.java#L163
and the target is determined here:
https://github.com/kohsuke/winp/blob/master/src/main/java/org/jvnet/winp/Native.java#L148

You can run through the `extractToStaticLocation` method as often, as you want, it will only extract the file once, as the target location is only depended on the passed in URL and that is the URL of the embedded DLL, that does not change at runtime.

**Comment by emilianbold** [comment]
+1

**Pull request merged**